examples/dam-app/src/index.js

```diff
@@ -0,0 +1,110 @@
+import './index.css';
+import { setup } from '@contentful/dam-app-base';
+import { EntityList } from '@contentful/f36-components';
+import pick from 'lodash/pick';
+import { useAsync } from 'react-async-hook';
+import { render } from 'react-dom';
+
+const CTA = 'Sample DAM Demo App';
+const FIELDS_TO_PERSIST = ['id', 'name', 'url'];
+
+setup({
+  cta: CTA,
+  name: 'Contentful DAM Demo App',
+  logo: 'https://images.ctfassets.net/fo9twyrwpveg/6eVeSgMr2EsEGiGc208c6M/f6d9ff47d8d26b3b238c6272a40d3a99/contentful-logo.png',
+  color: '#036FE3',
+  description:
+    'This is a sample Application to demonstrate how to make a custom DAM application on top of Contentful.',
+  parameterDefinitions: [
+    {
+      id: 'apiKey',
+      type: 'Symbol',
+      name: 'API Key',
+      description: 'Provide the API key here',
+      required: true,
+    },
+    {
+      id: 'projectId',
+      type: 'Number',
+      name: 'Project Id',
+      description: 'Provide the Project Id here',
+      required: true,
+    },
+  ],
+  validateParameters,
+  makeThumbnail,
+  renderDialog,
+  openDialog,
+  isDisabled: () => false,
+});
+
+function DialogLocation({ sdk }) {
+  const apiKey = sdk.parameters.installation.apiKey;
+  const projectId = sdk.parameters.installation.projectId;
+
+  const { result: damData } = useAsync(async () => {
+    const response = await fetch(
+      `/dam_api_response.json?api_key=${apiKey}&project_id=${projectId}`
+    );
+    return response.json();
+  }, [apiKey, projectId]);
```

Comment:
Do we really want to be so opinionated here and adding this whole library here? Why not just do a simple fetch in a `useEffect` ?

Reply:
I don't use `useEffect` for async methods as it doesn't handle unmounting and then tries to update a state of an unmounted component.
But for this example, I'll drop it.

Reply:
I agree, but maybe for the sake of the keeping the example simple we can be okay with that